src/configure/gen_c_primitives.ml

```diff
@@ -90,12 +98,24 @@ let prelude = "\
 #include <stdint.h>
 #include <stdbool.h>
 #include <inttypes.h>
+#define __STDC_WANT_IEC_60559_TYPES_EXT__
+#include <float.h>
 #include <caml/mlvalues.h>
 
 #define alignof(T) (offsetof(struct { char c; T t; }, t))
 #define STRINGIFY1(x) #x
 #define STRINGIFY(x) STRINGIFY1(x)
 
+#ifdef FLT16_MAX
```

Comment:
Note: we reuse the same detection method as in `ctypes_float16_availability` . These should be kept in sync